language/golang/develop.md

```diff
@@ -0,0 +1,475 @@
+		return listEntries(db, c)
+	})
+	e.POST("/add", func(c echo.Context) error {
+		return addEntry(db, c)
+	})
+
+	// Blast off!
+	port = os.Getenv("GODOCKER_PORT")
+	if port == "" {
+		port = "8000"
+	}
+	e.Logger.Fatal(e.Start(":" + port))
+}
+
+// listEntries returns a full copy of the store in JSON format to the client.
+func listEntries(db *sql.DB, c echo.Context) error {
+
+	rows, err := db.Query("SELECT key, value FROM store")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer rows.Close()
+
+	// Temporary storage for the values read from the DB
+	store := map[string]string{}
+
+	for rows.Next() {
+		var key, value string
+		if err := rows.Scan(&key, &value); err != nil {
+			log.Fatal(err)
+		}
+		store[key] = value
+	}
+
+	return c.JSON(http.StatusOK, store)
+}
+
+// addEntry adds a new entry to the key-value store
+// and returns the value in JSON format to the client.
+func addEntry(db *sql.DB, c echo.Context) error {
+
+	e := new(Entry)
+	if err := c.Bind(e); err != nil {
+		return err
+	}
+
+	err := crdb.ExecuteTx(context.Background(), db, nil, func(tx *sql.Tx) error {
+		if _, err := tx.Exec(
+			"INSERT INTO store (key, value) VALUES ($1, $2) ON CONFLICT (key) DO UPDATE SET value = excluded.value",
+			e.Key, e.Value); err != nil {
+			return err
+		}
+		return nil
+	})
+
+	if err != nil {
+		return c.JSON(http.StatusInternalServerError, err)
+	}
+
+	return c.JSON(http.StatusOK, e)
+}
+```
+
+Depending on the IDE you use, you might need to `go get` the modules imported in `main.go`:
+
+```shell
+$ go get github.com/cockroachdb/cockroach-go/crdb
+$ go get github.com/lib/pq
+```
+
+Since our application source code has been updated, we now need to rebuild our Docker image to reflect these changes.
+
+### Build the application
+
+We can build an updated image with the following command.
+
+```shell
+$ docker build --tag go-docker .
+```
+
+### Run the application
+
+Now, let’s run our container. But this time we’ll need to set quite a few environment variables so that our application could make the connection string necessary to access the database. We’ll do this right in the `docker run` command.
+
+Note, that in the following command the value of `PGUSER` must be set to the same value that you had picked before for the _Configure the database engine_ section. In our example, we use the value of `wumpus`.
+
+```shell
+$ docker run \
+  -it --rm -d \
+  --network godockernet \
+  --name rest-server \
+  -p 8000:8000 \
+  -e PGUSER=wumpus -e PGPASSWORD=passwd9 \
+  -e PGHOST=roach1 -e PGPORT=26257 -e PGDATABASE=godocker \
+  go-docker
+```
+
+### Testing the application
+
+Let’s test that our application is connected to the database and is able to save key-value pairs. We'll reuse the test input from the [Build your Go image](build-images.md) section:
+
+```shell
+$ curl --request POST \
+  --url http://localhost:8000/add
+  --header 'content-type: application/json'
+  --data '{"key": "name", "value": "Docker"}'
```

Comment:
Backslashes are missing.

```suggestion
$ curl --request POST \
  --url http://localhost:8000/add \
  --header 'content-type: application/json' \
  --data '{"key": "name", "value": "Docker"}'
```

Reply:
Yes, how odd that I've missed them... Thank you! Now it's fixed.